.github/workflows/libc-fullbuild-tests.yml

```diff
@@ -0,0 +1,76 @@
+# This workflow is for pre-commit testing of the LLVM-libc project.
+name: LLVM-libc Pre-commit Fullbuild Tests
+
+on:
+  pull_request:
+    branches: [ "main" ]
+    paths:
+      - 'libc/**'
+      - '.github/workflows/libc-fullbuild-tests.yml'
+
+jobs:
+  build:
+    runs-on: ubuntu-24.04
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - c_compiler: clang
+            cpp_compiler: clang++
+          # TODO: add back gcc build when it is fixed
+          # - c_compiler: gcc
+          #   cpp_compiler: g++
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Setup ccache
+      uses: hendrikmuhs/ccache-action@v1.2
+      with:
+        max-size: 1G
+        key: libc_fullbuild_${{ matrix.c_compiler }}
+        variant: sccache
+    
+    - name: Prepare dependencies (Ubuntu)
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y libmpfr-dev libgmp-dev libmpc-dev ninja-build linux-headers-generic linux-libc-dev
+        sudo ln -sf /usr/include/$(uname -p)-linux-gnu/asm /usr/include/asm
```

Comment:
Why do we need the symlink?

Reply:
Debian and its variants has a multilib sysroot, so arch-dependent headers are not installed into `/usr/include/`.

Ideally, `libc` should also add `/usr/include/$(uname -p)-linux-gnu/` in its include directories but we do not do that.